extension/js/content_scripts/webmail/gmail/gmail-webmail-startup.ts

```diff
@@ -0,0 +1,148 @@
+/* ©️ 2016 - present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com */
+
+import { Gmail } from '../../../common/api/email-provider/gmail/gmail';
+import { Env } from '../../../common/browser/env';
+import { Time } from '../../../common/browser/time';
+import { ClientConfiguration } from '../../../common/client-configuration';
+import { Str } from '../../../common/core/common';
+import { Injector } from '../../../common/inject';
+import { MessageRenderer } from '../../../common/message-renderer';
+import { Notifications } from '../../../common/notifications';
+import { Catch } from '../../../common/platform/catch';
+import { RelayManager } from '../../../common/relay-manager';
+import { XssSafeFactory } from '../../../common/xss-safe-factory';
+import { WebmailVariantObject, contentScriptSetupIfVacant } from '../generic/setup-webmail-content-script';
+import { WebmailElementReplacer } from '../generic/webmail-element-replacer';
+import { GmailElementReplacer } from './gmail-element-replacer';
+
+export class GmailWebmailStartup {
+  private replacer: WebmailElementReplacer;
+
+  public asyncConstructor = async () => {
+    this.injectFCVarScript();
+    await Time.sleep(100); // Wait until injected dom is added
+    const webmailVariant = this.determineWebmailVariant();
+    await contentScriptSetupIfVacant({
+      name: 'gmail',
+      variant: webmailVariant.gmailVariant,
+      getUserAccountEmail: () => this.getUserAccountEmail(webmailVariant),
+      getUserFullName: () => $('div.gb_hb div.gb_lb').text() || $('div.gb_Fb.gb_Hb').text(),
+      getReplacer: () => this.replacer,
+      start: this.start,
+    });
+  };
+
+  private start = async (
+    acctEmail: string,
+    clientConfiguration: ClientConfiguration,
+    injector: Injector,
+    notifications: Notifications,
+    factory: XssSafeFactory,
+    notifyMurdered: () => void,
+    relayManager: RelayManager
+  ) => {
+    this.hijackGmailHotkeys();
+    injector.btns();
+    const messageRenderer = await MessageRenderer.newInstance(acctEmail, new Gmail(acctEmail), relayManager, factory);
+    this.replacer = new GmailElementReplacer(factory, clientConfiguration, acctEmail, messageRenderer, injector, notifications, relayManager);
+    await notifications.showInitial(acctEmail);
+    this.replacer.runIntervalFunctionsPeriodically();
+  };
```

Comment:
`notifyMurdered` parameter isn't used here.
I think we can make it abstract method of `WebmailElementReplacer`, so it'll be implemented directly in `GmailElementReplacer`, and later in `ThunderbirdElementReplacer`

Reply:
Good call that it's unused here 👍 

I imagine it would be reused as is in both Thunderbird and Gmail as is, since it particularly relates to Firefox environment and Thunderbird is based on Firefox, so it may get triggered the same there. I'll see if I can clean this up.

Reply:
Removed the unused param